src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/JintLuceneDocumentConverter.cs

```diff
@@ -346,32 +362,47 @@ private static JsValue TryDetectDynamicFieldCreation(string property, ObjectInst
                     var propertyNameAsString = propertyName.AsString();
                     if (string.Equals(propertyNameAsString, nameof(CreateFieldOptions.Indexing), StringComparison.OrdinalIgnoreCase))
                     {
-                        field.Indexing = GetEnum<FieldIndexing>(optionValue, propertyNameAsString);
+                        fieldIndexing = GetEnum<FieldIndexing>(optionValue, propertyNameAsString);
 
                         continue;
                     }
 
                     if (string.Equals(propertyNameAsString, nameof(CreateFieldOptions.Storage), StringComparison.OrdinalIgnoreCase))
                     {
                         if (optionValue.IsBoolean())
-                            field.Storage = optionValue.AsBoolean()
+                            fieldStorage = optionValue.AsBoolean()
                                 ? FieldStorage.Yes
                                 : FieldStorage.No;
                         else
-                            field.Storage = GetEnum<FieldStorage>(optionValue, propertyNameAsString);
+                            fieldStorage = GetEnum<FieldStorage>(optionValue, propertyNameAsString);
 
                         continue;
                     }
 
                     if (string.Equals(propertyNameAsString, nameof(CreateFieldOptions.TermVector), StringComparison.OrdinalIgnoreCase))
                     {
-                        field.TermVector = GetEnum<FieldTermVector>(optionValue, propertyNameAsString);
+                        termVector = GetEnum<FieldTermVector>(optionValue, propertyNameAsString);
 
                         continue;
                     }
                 }
             }
 
+            field = IndexField.Create(fieldName, 
+                new IndexFieldOptions()
+                {
+                    Indexing = fieldIndexing, 
+                    Storage = fieldStorage, 
+                    TermVector = termVector
+                }, _allFields);
+
+            if (_dynamicFieldsDynamicAnalyzer)
+            {
+                //This will get analyzer from dynamic field configuration.
+                CurrentIndexingScope.Current.DynamicFields ??= new();
+                CurrentIndexingScope.Current.DynamicFields[fieldName] = field;
```

Comment:
this will blindly substitute, I guess it assumes that it was check beforehand? please confirm

Reply:
This is exact copy of current flow inside Lucene C# CreateField:
https://github.com/ravendb/ravendb/blob/7476d9de3a31c1fe66a4af98271a01b08e15e42c/src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs#L493